pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1206,15 +1206,24 @@
       ?~  maybe-channel=(~(get by session.channel-state.state) channel-id)
         %^  return-static-data-on-duct  404  'text/html'
         (error-page 404 %.y url.request ~)
-      ::  if there's already a duct listening to this channel, we must 400
-      ::
-      ?:  ?=([%| *] state.u.maybe-channel)
-        %^  return-static-data-on-duct  400  'text/html'
-        (error-page 400 %.y url.request "channel already bound")
       ::  when opening an event-stream, we must cancel our timeout timer
-      ::
-      =.  moves
-        [(cancel-timeout-move channel-id p.state.u.maybe-channel) moves]
+      ::  if there's no duct already bound. Else, kill the old request
+      ::  and replace it
+      ::
+      =^  cancel-moves  state
+        ?.  ?=([%| *] state.u.maybe-channel)  
+          :_  state
+          (cancel-timeout-move channel-id p.state.u.maybe-channel)^~
+        =/  cancel-heartbeat  
+          ?~  heartbeat.u.maybe-channel  ~
+          :_  ~
+          %+  cancel-heartbeat-move  channel-id 
+          [date duct]:u.heartbeat.u.maybe-channel
+        =-  [(weld cancel-heartbeat -<) ->]
+        %.  [%cancel ~]
+        %*  .  handle-response
+          duct  p.state.u.maybe-channel
+        ==
```

Comment:
Does `handle-response(duct p.state.u.maybe-channel)` (or `%_` if you must go tall) not work?

Reply:
It does, but i've had some issue doing this in the past for whatever reason, updated!